scripts/utils/migrate/create-index-pages.js

```diff
@@ -1,96 +1,127 @@
+  [/new relic/gi, 'New Relic'],
+  [/\bphp\b/gi, 'PHP'],
+  [/\bui\b/gi, 'UI'],
+  [/\bapi\b/gi, 'API'],
+  [/([wW])hats/gi, "$1hat's"],
+];
+
+const replace = (string) =>
+  REPLACEMENTS.reduce(
+    (str, [regex, replacement]) => str.replace(regex, replacement),
+    string
+  );
+
+const capitalize = (word) => word.charAt(0).toUpperCase() + word.slice(1);
+
+const sentenceCase = (dirname) => {
+  const [firstWord, ...words] = dirname.split('-');
+  const title = [capitalize(firstWord), ...words].join(' ');
+
+  return replace(title);
 };
 
-const createIndexPage = async (dir, level = 2) => {
-  // dont run for the attribute definitions
-  if (dir === path.join(BASE_DIR, 'attribute-dictionary')) {
-    return false;
-  }
-
-  const title = getTitle(dir);
+const SKIPPED_FOLDERS = ['src/content/attribute-dictionary'];
 
-  try {
-    const nodes = (await fs.readdir(dir, { withFileTypes: true })) || [];
+const shouldSkipDirectory = (dir) => SKIPPED_FOLDERS.includes(dir.path);
 
-    // get files (and links) for this directory
-    const files = nodes.filter((node) => !node.isDirectory());
-    const dirLinks = getPageContent(dir, files);
+const toURL = (node) =>
+  path.join(
+    '/',
+    node.path
+      .replace('.mdx', '')
+      .replace(BASE_DIR, '')
+      .replace(/\/index\/?$/, '')
+  );
 
-    // get all sub directories
-    const subDirs = nodes
-      .filter((node) => node.isDirectory())
-      .map(({ name }) => path.join(dir, name));
-
-    // get links for sub directories (make index pages along the way)
-    const subLinks = await subDirs.reduce(async (contentPromise, subDir) => {
-      // Due to this being async, the previous (accumulator) result is a promise.
-      // We need to await the results of the last loop, before we can add to it.
-      const content = await contentPromise;
-
-      const subDirContent = await createIndexPage(subDir, level + 1);
-      const subDirHeading = getSubheading(subDir, level);
-
-      return [content, subDirHeading, subDirContent].join('\n');
-    }, '');
-
-    // if there is an index.mdx, don't overwrite or make a new one
-    if (files.some((file) => file.name === 'index.mdx')) return;
-
-    // combine links and add them to a new index.mdx file
-    const fileName = `${dir}/index.mdx`;
-    const content = getFrontmatter(title) + dirLinks + subLinks;
-    await fs.writeFile(fileName, content);
+const depthOf = (node, dir) => {
+  return node.path.replace(new RegExp(`${dir.path}\\/`, '')).split('/').length;
+};
 
-    // return the links for this page (to be used by a parent, if applicable)
-    return dirLinks + subLinks;
-  } catch (err) {
-    logger.error(`Unabled to create index for ${dir}: ${err}`);
-  }
+const generateMDX = (dir) => {
+  const tree = root([
+    frontmatter({ title: sentenceCase(dir.basename), template: 'basicDoc' }),
+  ]);
+
+  visit(
+    dir,
+    (node) => node !== dir,
+    (node) => {
+      if (shouldSkipDirectory(node)) {
+        return [visit.SKIP];
```

Comment:
How does this exactly work? `visit.SKIP`

Reply:
This essentially tells the utility to stop traversing the subtree if this is returned. If I had a normal `return` here, all that does is avoid processing the `node` we are on, but the visitor will continue to traverse it's subtree.

In this case, I don't want to visit any of the nodes in the subtree when we are in the directory `attribute-dictionary`, so this tells it to skip traversing it. Does that make sense? 

Here is the documentation on this particular bit: https://github.com/syntax-tree/unist-util-visit-parents#returns

Reply:
Yes! Thank you. I didn't really understand what it was, but it appears that it's an attribute on the object that's a string (`'skip') where when you return it, it will know to skip that node. Is that correct?

Reply:
Thats correct!